jobrunner/create_or_update_jobs.py

```diff
@@ -299,6 +304,16 @@ def assert_new_jobs_created(job_request, new_jobs, current_jobs):
     raise Exception(f"Unexpected job states after scheduling: {current_job_states}")
 
 
+def assert_codelists_ok(job_request, new_jobs):
+    if job_request.codelists_ok:
+        return True
+    for job in new_jobs:
+        # Codelists are out of date; fail the entire job request if any job
+        # requires database access
+        if job.requires_db:
+            raise JobRequestError("Codelists are out of date")
```

Comment:
Is it worth adding any more detail here? I guess the user will have the warning message about out-of-date codelists already at the top of the page, won't they?

Reply:
My thought was that they'd probably have already seen the out of date codelist message and ignored it - or at least would have seen it enough times before to know what this meant.  But I've assumed that before and been wrong.
Since it fails on a specific job, and we assume that's a dependency that they didn't explicitly request (otherwise job-server would have refused to create the job request in the first place) it could say something like "Codelists are out of date (required by Job X)"?